bytecomp/translmod.ml

```diff
@@ -656,8 +641,30 @@ and transl_structure loc fields cc rootpath final_env = function
                transl_module Tcoerce_none None modl, body),
           size
 
+      | Tstr_open od ->
+          begin match od.open_expr.mod_desc with
+          | Tmod_ident _ ->
+            transl_structure loc fields cc rootpath final_env rem
+          | _ ->
```

Comment:
It'd be good to treat `open struct ... end` specially here to avoid the extra allocation. Same applies to `include` really but I guess that pattern is less common.

Reply:
I suspect the reason `include struct ... end` is not treated specially here is because it's actually annoying to relate the the internal idents to the external ones.
The same is true in the `open struct ... end` case, so I'll delay the special handling for now until I a clean solution which can handle both constructs nicely.